src/connection.toit

```diff
@@ -149,7 +151,11 @@ class Connection:
     headers := read_headers_
     body_reader := body_reader_ headers --request=true
 
-    current_reader_ = body_reader
+    if body_reader is ContentLengthReader and (body_reader as ContentLengthReader).size == 0:
```

Comment:
This feels a little bit odd to me. Why do we only do this check when --request=true? Is this a convenience so you don't have to read the stuff on the server side? I guess most clients will read responses, but we worry that we will forget to drain the requests? Doesn't that happen automatically anyway? 

If we want to stick with this, maybe `body_reader_` should just return null in this case instead. It feels like that's where we want to make the decision?

Reply:
I changed body_reader_ to return null, and a few changes to keep things working.